src/test/java/com/google/firebase/remoteconfig/TemplateTest.java

```diff
@@ -59,4 +67,77 @@ public void testSetNullParameterGroups() {
     Template template = new Template();
     template.setParameterGroups(null);
   }
+
+  @Test
+  public void testEquality() {
```

Comment:
I'd advise splitting this into multiple smaller test cases. You can define some class-level constants for values that should be shared across test cases.

Reply:
I use all the template instances created here in a series of `assertNotEquals()` at the end of the test... would it be fine to move all those instances to class level?